### DISCUSSION TITLE
Cannot Reference DataflowStartFlexTemplateOperator in AF 2.3.2 but can reference it in AF 2.3.1

### DISCUSSION THREAD

**Discussion by myang-clgx** [apache/airflow#24273] [General]
### Apache Airflow Provider(s)

google

### Versions of Apache Airflow Providers

apache-airflow-providers-apache-beam 3.4.0
apache-airflow-providers-ftp         2.1.2
apache-airflow-providers-google      7.0.0
apache-airflow-providers-http        2.1.2
apache-airflow-providers-imap        2.2.3
apache-airflow-providers-sqlite      2.1.3

### Apache Airflow version

2.3.2 (latest released)

### Operating System

MAC OSX 12.4

### Deployment

Other

### Deployment details

Local deployment without Docker

### What happened

no reference to DataflowStartFlexTemplateOperator because it cannot find "google" under airflow.providers

### What you think should happen instead

It was referenced in AF 2.3.1 and now it cannot.

### How to reproduce
0. have an AF DAG only contains DataflowStartTemplateOperator code 
(from airflow.providers.google.cloud.operators.dataflow import DataflowStartFlexTemplateOperator)
1. Install AF 2.3.2 in terminal via pip install after creating a new Conda Virtual Environment
2. Re-select the appropriate Conda environment if you create a new Conda Virtual Environment
3. See the error after PyCharm reindex and getting those libraries dependencies
4. Then switch the Conda environment that contains AF 2.3.1
5. Observe to see if there is any error after PyCharm reindex and getting those libraries dependencies and there is no error on mis-reference.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
I think you should explain what the problem is in detail. I do not understand what kind of errors you have (you did not explain what is missing), But you are likely installing airflow wrongly. As explained in https://airflow.apache.org/docs/apache-airflow/stable/installation/index.html#using-pypi - the ONLY official way of installing Airflow is using PIP and constraints. Conda has no problem with using `pip` to install airlfow in Conda virtual environments, so likely your problem comes from installing airflwo without constraints and you can fix it by using them.

**Reply by myang-clgx** [reply]
I installed airflow by using pip install under Conda created virtual environment.

The error that I am seeing right now is that PyCharm does not recognize google module under airflow.providers and it cannot recognize/reference "DataflowStartFlexTemplateOperator" from import above the existing DAG

The problem detail was in How to reproduce section.

**Reply by potiuk** [reply]
Well, I think you should configure your PyCharm correctly. And. I do not know how. I do not know Conda, I do not know how it interacts with scanning the repo and finding (or not) modules. Note that airflow providers use implicit namespaces, so that might be your problem - but this is something you have to figure yourself out. 

If you find out how, it would be great if you mention it here. We cannot be responsible for your way of building development environment. 

if you look at our CONTRIBUTING doc - we have quick start that covers step-by-step guide on what to do when you want to have Airflow dev env working (and it works - I am using it all the time). https://github.com/apache/airflow/blob/main/CONTRIBUTORS_QUICK_START_PYCHARM.rst - if you follow it things will work for Airflow itself. Just try it - then you can start comparing your env with the one that you created with Conda and maybe you will find some differences. 

This is as close as it can get to helping you to solve your problem.

**Reply by potiuk** [reply]
BtW. I read your "reproduction" steps and I still, no matter how much time I re-read it, do not understand what your reproduction looks like. There are some mental shortcuts there which I completely do not understand. If you **still** want some help I highly recommend to be as precise as you can on what you do - possibly even providing some screenshots along the way.

Help us to help you. If you need help make sure that there is no extra work on the side of people who are reading it and try to help you for free - simply spend some time on clearly and precisely explaining what you try to do - in the way that people reading it will not have to spend time on figuring out what you **really** wanted to do. It's you who are reaching for help and we try to help in our free time in the software you get for free, so please spend your time there if you need help so that those who want to help don't have to do it.

**Comment by potiuk** [comment]
Converted it into discussion anyway.
